content/articles/gulp4-primer.md

```diff
@@ -0,0 +1,153 @@
+```
+The boilerplate demo tests are going to fail at that point because of changes in the api of gulp and probably some in other libraries we just updated as well.  
+
+You can see the commit [here](https://github.com/Keats/ng-boilerplate/commit/49a339cc5cbde8f5374b8f4c915a548dc2916cc4).
+
+## Migrating the gulpfile
+### Calling gulp
+The first thing we need to do is ensure that we are using the right version of gulp.  You might have another version installed globally or nothing installed globally like me.  
+The easiest way (at least to me) to deal with that issue is the [npm scripts](https://docs.npmjs.com/misc/scripts) part of the package.json.
+
+```js
+// package.json
+"scripts": {
+  "gulp": "./node_modules/gulp/bin/gulp.js"
+},
+
+// in your terminal, instead of using gulp, use npm run gulp
+npm run gulp
+```
+
+The downside is that you will get npm errors instead of the regular one but that's only an issue when building your gulpfile.
+Let's not forget to update the travis.yml to use that npm script too.  
+
+You can see the commit [here](https://github.com/Keats/ng-boilerplate/commit/ef30315f43580e8b3c1ba85d4536c7fe0d69365d).
+
+### Fixing the errors we're getting
+If you run the gulpfile locally or look at the [travis build](https://travis-ci.org/Keats/ng-boilerplate/builds/54252789), you will notice we get the following error:
+```bash
+assert.js:87
+  throw new assert.AssertionError({
+        ^
+AssertionError: Task function must be specified
+    at Gulp.set (/home/vincent/Code/ng-boilerplate/node_modules/gulp/node_modules/undertaker/lib/set.js:14:3)
+    at Gulp.task (/home/vincent/Code/ng-boilerplate/node_modules/gulp/node_modules/undertaker/lib/task.js:14:8)
+....
+```
+This comes from the fact that gulp.task changed and the 3 parameters signature was removed.
+In short, tasks that look like the one below will need to be changed to remove the `[]` parameter:
+```js
+gulp.task('default', ['build'], function () {
+  return runSequence(['watch', 'karma-watch']);
+});
+```
+To give a quick background to people not familiar to gulp, the list parameter is a list of tasks that need to be run before running that task.
+In this case it will call the build task before running watch and karma-watch in order (runSequence is another thing that can be changed but we will get back to that later).
+To solve that very common issue of task dependencies, two methods were added:
+
+- `gulp.series`: will run the tasks in order
+- `gulp.parallel`: will run the tasks in parallel
+
+Let's fix our error by replacing those dependent tasks, for example for the default task:
+
+```js
+gulp.task(
+  'default',
+  gulp.series('build', gulp.parallel('browser-sync', 'watch', 'karma-watch'))
+);
+```
+
+In that case we first want to run build, and then browser-sync, watch and karma-watch in parallel. Easy (the commit below will contain a buggy version since I did put browser-sync in the series by mistake, *easy* right).  
```

Comment:
meant to say that the commit i did for that part has a bug since i made a mistake